src/shell.zig

```diff
@@ -857,3 +857,56 @@ fn discover_project_root() !std.fs.Dir {
 
     return error.DiscoverProjectRootDepthExceeded;
 }
+
+pub const HttpOptions = struct {
+    pub const ContentType = enum {
+        json,
+
+        fn string(content_type: ContentType) []const u8 {
+            return switch (content_type) {
+                .json => "application/json",
+            };
+        }
+    };
+
+    content_type: ?ContentType,
+    authorization: ?[]const u8,
+};
+
+/// Issues an HTTP POST request with the given `body` to the given `url`.
+///
+/// If the response is not 200 OK, the response body is logged and an error is returned.
+pub fn http_post(shell: *Shell, url: []const u8, body: []const u8, options: HttpOptions) !void {
+    errdefer |err| log.err("failed to HTTP post to \"{s}\": {s}", .{ url, @errorName(err) });
+
+    const allocator = shell.arena.allocator();
+    var client = std.http.Client{ .allocator = allocator };
+    defer client.deinit();
```

Comment:
Let's use `shell.gpa` here --- arena is for caller, such that they don't have to think about one-off memory management. But _inside_ shell we are writing re-usable functions, so it's relatively easier to get memory management right.

So, we could be using `gpa` here and be careful around defers. 


If, at some point, we would love to _return_ some data, then the returned data (and only it) would be allocated in the arena.